nothunks.cabal

```diff
@@ -54,6 +59,8 @@ library
       build-depends:  text       >= 1.2  && < 1.3 || >= 2 && < 2.2
     if flag(vector)
       build-depends:  vector     >= 0.12 && < 0.14
+    if flag(wherefrom)
+      build-depends:  wherefrom-compat >= 0.1.1.0 && < 0.2.0.0
```

Comment:
I'd just use it by default, it will be simpler user experience without flags.

Reply:
I was originally going to do that but I think that would entail raising the minimum base bound to 4.16. If you're happy with that I'll drop the flag and I'll probably be able to eliminate some CPP too

Reply:
I am fine with that.

Reply:
let me be more precise, in the `*.cabal` file:
```
if impl(ghc >= 9.2)
  build-depends: wherefrom-compat ^>= 0.1.1
```
And then we can use CPP in the haskell module to only import / use `wherefrom` in the right `ghc` version.

Reply:
Ah i see I hadn't thought of that I'll implement that